packages/docid/src/index.ts

```diff
@@ -0,0 +1,311 @@
+   *
+   * @returns {string}
+   * @readonly
+   */
+  get subnetName (): string {
+    const name = getKey(subnets, this._subnet)
+    if (!name) throw new Error('subnetName: no registered name available')
+    return name 
+  }
+
+  /**
+   * Get multibase name string
+   *
+   * @returns {string}
+   * @readonly
+   */
+  get multibaseName (): string {
+    return this._multibaseName
+  }
+
+   /**
+   * Get multihash bytes of DocId
+   *
+   * @returns {Uint8Array}
+   * @readonly
+   */
+  get multihash (): Uint8Array {
+    return this._cid.multihash
+  }
+
+   /**
+   * Get CID codec name string
+   *
+   * @returns {string}
+   * @readonly
+   */
+  get codec (): string {
+    return this._cid.codec
+  }
+
+  /**
+   * Get bytes of DocId
+   *
+   * @returns {Uint8Array}
+   * @readonly
+   */
+  get bytes (): Uint8Array {
+    if (this._bytes == null) {
+      const codec = varint.encode(DOCID_CODEC)
+      const subnet = varint.encode(this._subnet)
+      const doctype = varint.encode(this._doctype)
+      const versionBytes =  this._version ? this._version.bytes : new Uint8Array(0)
+      this._bytes = uint8ArrayConcat([
+        codec, subnet, doctype, this._cid.bytes, versionBytes
+      ])
+    }
+    return this._bytes
+  }
+
+ /**
+   * Compare equality with another DocID.
+   *
+   * @param   {DocID}   other
+   * @returns {bool}
+   */
+  equals(other: DocID | Uint8Array | string): boolean {
+    let otherDocID
+    if (typeof other === 'string')
+      otherDocID = DocID.fromString(other)
+    else if (other instanceof Uint8Array) {
+      otherDocID = DocID.fromBytes(other)
+    } else {
+      otherDocID = other
+    }
+
+    return this.type === otherDocID.type &&
+      this.subnet === otherDocID.subnet &&
+      this.cid.equals(otherDocID.cid)
+  }
+
+  /**
+   * Encode the DocID into a string.
+   *
+   * @param   {string}   [base=this.multibaseName]   Base encoding to use.
+   * @returns {string}
+   */
+  toBaseEncodedString (base: string = this.multibaseName): string {
+    return uint8ArrayToString(multibase.encode(base, this.bytes))
+  }
+
+  /**
+   * Encode the DocID into a string.
+   *
+   * @param   {string}   [base=this.multibaseName]   Base encoding to use.
+   * @returns {string}
+   */
+  toString(base?: string): string {
+    return this.toBaseEncodedString(base)
+  }
+
+  /**
+   * Encode the DocID into a url
+   *
+   * @param   {string}   [base=this.multibaseName]   Base encoding to use.
+   * @returns {string}
+   */
+  toUrl(base: string): string {
+    return `ceramic://${this.toBaseEncodedString(base)}`
+  }
+
+  /**
+   * DocId(k3y52l7mkcvtg023bt9txegccxe1bah8os3naw5asin3baf3l3t54atn0cuy98yws)
+   *
+   * @returns {String}
+   */
+  [Symbol.for('nodejs.util.inspect.custom')] (): string {
+    return `DocID(${this.toString()})`
+  }
+
+  [Symbol.toPrimitive](hint: string): string | Uint8Array {
+    if (hint === 'string') return this.toString()
+    if (hint === 'default') return this.bytes
```

Comment:
I think the primitives can only be `string` or `number`, here I don't think it's necessary to use the `hint`, could simply be:
```ts
[Symbol.toPrimitive](): string {
  return this.toString()
}
```

Reply:
it does have default - https://tc39.es/ecma262/#sec-toprimitive but maybe should also be string, assumed most string context would be explicit and default would be used for like []+docid, so bytes seemed reasonable

Reply:
Yes sorry I meant for the returned type, I'm not sure `Uint8Array` is valid because it's not a JS primitive.
At least my understanding for this method is that a primitive is always `string` or `number`, and the `hint` is there to give contextual information, cf https://javascript.info/object-toprimitive
Maybe it's not even needed to implement this method, assuming the JS engine defaults to using `toString()` though I don't know if it is standard behavior.

Reply:
ah i see what your saying! yeah so just return string seems to make sense and we will make it explicit still